### PR TITLE
Implement StringSemanticEquals for GoDuration

### PR DESCRIPTION
**Pull request by czeltser**

### CODE REVIEW

File: timetypes/go_duration_value.go [hunk 1]
```diff
@@ -106,6 +106,30 @@ func (d GoDuration) ValueGoDuration() (time.Duration, diag.Diagnostics) {
...
+func (d GoDuration) StringSemanticEquals(ctx context.Context, newValuable basetypes.StringValuable) (bool, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	newValue, ok := newValuable.(GoDuration)
+	if !ok {
+		diags.AddError(
+			"Semantic Equality Check Error",
+			"An unexpected value type was received while performing semantic equality checks. "+
+				"Please report this to the provider developers.\n\n"+
+				"Expected Value Type: "+fmt.Sprintf("%T", GoDurationType{})+"\n"+
```
Comment (austinvalle) [inline timetypes/go_duration_value.go:120]: We want the value type here (`GoDuration`), so we can just use the receiver value itself
```suggestion
				"Expected Value Type: "+fmt.Sprintf("%T", d)+"\n"+
```

File: timetypes/go_duration_value_test.go [hunk 2]
```diff
@@ -175,6 +175,64 @@ func TestDurationValidateParameter(t *testing.T) {
...
+		"error - not a Duration value": {
+			currentDuration: timetypes.NewGoDurationValueFromStringMust("56s"),
+			givenDuration:   basetypes.NewStringValue("abcdef"),
+			expectedMatch:   false,
+			expectedDiags: diag.Diagnostics{
+				diag.NewErrorDiagnostic(
+					"Semantic Equality Check Error",
+					"An unexpected value type was received while performing semantic equality checks. "+
+						"Please report this to the provider developers.\n\n"+
+						"Expected Value Type: timetypes.GoDurationType\n"+
```
Comment (austinvalle) [inline timetypes/go_duration_value_test.go:211]: If we update the err msg above, I believe this should be the new message
```suggestion
						"Expected Value Type: timetypes.GoDuration\n"+
```

**Review by austinvalle — COMMENTED** [review]
Thanks for the contribution @czeltser! All the changes here look good, just one small note about the error message.

Can you add a changelog for your update here as well? I think it makes sense to classify this as an `ENHANCEMENT` and we can release it with `v0.5.0`
https://github.com/hashicorp/terraform-plugin-framework-timetypes/blob/main/.github/CONTRIBUTING.md#changelog

**Comment by czeltser** [comment]
Thank you for the review @austinvalle! I've updated the MR with your suggestions. I pushed one additional change which I missed the first time around to remove the semantic equality logic from `Equals()`, as recommended in the note [here](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/types/custom#value-type).

**Review by austinvalle — APPROVED** [review]
> I pushed one additional change which I missed the first time around to remove the semantic equality logic from `Equals()`, as recommended in the note [here](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/types/custom#value-type).

Great catch! Everything looks good here 🚀 , so I'll make a note to release `terraform-plugin-framework-timetypes v0.5.0` sometime next week.

If you'd like to test the changes in the meantime, you can always pull from `main`:
```bash
go get -u github.com/hashicorp/terraform-plugin-framework-timetypes@main
```

Thanks!

**Pull request merged**